lib/rufus/lua/lib_lua_5_1.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Rufus::Lua
+  module Lib
+    module Lua5_1
+      def self.included(mod)
+        mod.attach_function :lua_call, [ :pointer, :int, :int ], :void
+
+        mod.attach_function :lua_pcall, [ :pointer, :int, :int, :int ], :int
+
+        mod.attach_function :lua_tonumber, [ :pointer, :int ], :double
+
+        mod.attach_function :lua_remove, [ :pointer, :int ], :void
+
+        mod.attach_function :luaL_loadbuffer, [ :pointer, :string, :int, :string ], :int
+
+        mod.attach_function :lua_objlen, [ :pointer, :int ], :int
+        mod.extend(ClassMethods)
+      end
+
+      LUA_REGISTRYINDEX = -10000
+      LUA_GLOBALINDEX = -10002
+
+      module ClassMethods
+        def lua_setglobal(pointer, name)
+          lua_setfield(pointer, LUA_GLOBALINDEX, name)
+        end
+
+        def lua_getglobal(pointer, name)
+          lua_getfield(pointer, LUA_GLOBALINDEX, name)
+        end
```

Comment:
These methods don't exist in Lua 5.1, so I added these helper methods since they accomplish the same thing